lib/galaxy/model/store/__init__.py

```diff
@@ -2420,6 +2420,15 @@ def _init_crate(self) -> ROCrate:
         for dataset, _ in self.included_datasets.values():
             if dataset.dataset.id in self.dataset_id_to_path:
                 file_name, _ = self.dataset_id_to_path[dataset.dataset.id]
+                if file_name is None:
```

Comment:
I'd make sure this is really in discarded state and not that this is a different bug we're hiding here.